packages/ketcher-core/src/application/render/renderers/BaseMonomerRenderer.ts

```diff
@@ -98,6 +98,36 @@ export abstract class BaseMonomerRenderer extends BaseRenderer {
     }
   }
 
+  public redrawAttachmentPointsCoordinates() {
+    const chosenAttachmentPointName =
+      this.monomer.chosenFirstAttachmentPointForBond;
+    const chosenAttachmentPoint = this.attachmentPoints.find(
+      (item) => item.getAttachmentPointName() === chosenAttachmentPointName,
+    );
+    const angle = chosenAttachmentPoint?.getAngle();
+    const allAngles = this.attachmentPoints.map((item) => {
+      return item.getAngle();
+    });
+    const isSectorOccupied = allAngles.some((item) => {
+      if (angle !== item && typeof angle === 'number') {
+        return Math.abs(angle - item) < 20 || Math.abs(angle - item) > 340;
+      }
+      return false;
+    });
+
+    if (isSectorOccupied) {
+      this.redrawAttachmentPoints();
+      return;
+    }
+
+    const attachmentPointName = this.monomer.chosenFirstAttachmentPointForBond;
+    const attachmentPoint = this.attachmentPoints.find(
+      (item) => item.getAttachmentPointName() === attachmentPointName,
+    );
+    if (!attachmentPoint || attachmentPointName === null) return;
```

Comment:
I would suggest use assert here because in this case attachment point should exist.

Reply:
Fixed